plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -100,6 +114,14 @@ public PinotMetadata(
                         }, executor));
 
         executor.execute(() -> this.allTablesCache.refresh(ALL_TABLES_CACHE_KEY));
+        this.maxRowsPerBrokerQuery = pinotConfig.getMaxRowsForBrokerQueries();
+        this.aggregateFunctionRewriter = new AggregateFunctionRewriter(identity(), ImmutableSet.<AggregateFunctionRule>builder()
+                .add(new ImplementCountAll())
+                .add(new ImplementAvg())
+                .add(new ImplementMinMax())
+                .add(new ImplementSum())
+                .add(new ImplementApproxDistinct())
```

Comment:
Also, do you have any idea how we can push down `count(distinct A)` to Pinot as `distinctCount(A)`  or  `count(distinct A)` ?

Note, Pinot internal will rewrite `count(distinct A)`  to `distinctCount(A)` as well.

Reply:
I have code I can put back as an additional commit - so the broker request should use `count(distinct A)` and not `distinctCount(A)`?

Reply:
Had a concern that it can use a lot of resources on the pinot servers though since it would have to scan all the segments that passed the filter and it would be done on a single broker vs many trino workers. wdyt?

Reply:
hmm, this is a good point.

I still feel that the Pinot side should handle the count(distinct ) queries since it would be much cheaper computation comparing to stream data to trino workers. Also pinot side has done many optimization for this type of query.

Also, we can add server and session configs to allow users disable pushdown for distinct count queries. So they can skip pushdown if the intermediate data are too large.

Reply:
Update: added a session property to control whether distinct count is pushed down, implemented distinct count pushdown and added tests.

Reply:
Last commit can be squashed into the "Implement aggregation pushdown" - kept it separate for now, hopefully easier to review. @martint, @findepi , @xiangfu0  - thanks so much for all the suggestions, ideas and learning!